### ISSUE TITLE
Loss becomes nan after some epochs

### ISSUE THREAD

**Issue by inejc** [inejc/paragraph-vectors#16]
##### PyTorch and CUDA versions
PyTorch: `0.2.0.post3`
CUDA: `8.0`

##### Description
Loss suddenly becomes nan (was somewhere between 0.4 and 0.5 in the previous epoch) when training on a dataset comprised of 74219 documents with vocabulary size 91417.

##### Additional info (stack trace, etc.)
Training on a GPU with the following parameters:
`--data_file_name 'arxiv_min.csv' --num_epochs 500 --batch_size 512 --context_size 4 --num_noise_words 10 --vec_dim 300 --lr 0.001 --max_generated_batches 500 --num_workers 8`
